src/plugins/spec/spec.c

```diff
@@ -578,7 +589,354 @@ static int handleErrors (Key * parentKey, KeySet * ks, Key * key, Key * specKey,
+	const char * relativeName = strchr (keyName (specKey), '/') + strlen (strchr (keyName (parentKey), '/')) + 1;
+
+	int hasWildcard = 0;
+	int singleWC = 0;
+	int multiWC = 0;
+	int arrayWC = 0;
+	int metaHasWildcard = 0;
+	char * ptr = (char *)relativeName;
+
+	// Counting occurrences of ?, _ and # in spec keyname
+	while (*ptr)
+	{
+		if (*ptr == '?')
+		{
+			++singleWC;
+		}
+		else if (*ptr == '_')
+		{
+			++multiWC;
+		}
+		else if (*ptr == '#')
+		{
+			++arrayWC;
+		}
+		++ptr;
+	}
+	hasWildcard = singleWC + multiWC + arrayWC;
+
+	// check if metastring has wildcards
+	if (hasWildcard)
+	{
+		ptr = (char *)metaString;
+		while (*ptr)
+		{
+			if (*ptr == '?' || *ptr == '_' || *ptr == '#') ++metaHasWildcard;
+			++ptr;
+		}
+	}
+
+	// both keyname and string have wildcards
+	if (metaHasWildcard && hasWildcard)
+	{
+		// keyname with wildcards matched to actual values relative to parent key
+		const char * relativeFullName = strchr (keyName (key), '/') + strlen (strchr (keyName (parentKey), '/')) + 1;
+
+		size_t newMetaSize = keyGetValueSize (metaKey) + 3 * (strlen (relativeFullName) - strlen (relativeName));
+		char * newMetaString = elektraCalloc (newMetaSize);
+		char * src = (char *)metaString;
+		char * dest = newMetaString;
+		int singleWCCounter = 0;
+		int multiWCCounter = 0;
+		int arrayWCCounter = 0;
+
+		// search and replace wildcards using the same order the globbing key has module the number of occurrences.
+		while (*src)
+		{
+			if (*src != '?' && *src != '_' && *src != '#')
+			{
+				*dest = *src;
+				++src;
+				++dest;
+			}
+			else
+			{
+				// found a wildcard
+
+				char * segmentStart = src;
+				char * segmentEnd = src;
+				int localCounter = 0;
+				ptr = (char *)relativeName;
+
+				// try to find the corresponding wildcard in the globbing key.
+				if (*src == '?' && singleWC)
+				{
+					++singleWCCounter;
+					if (singleWCCounter > singleWC) singleWCCounter = 1;
+					while (localCounter < singleWCCounter)
+					{
+						if (*ptr == '?') ++localCounter;
+						++ptr;
+					}
+				}
+				else if (*src == '_' && multiWC)
+				{
+					++multiWCCounter;
+					if (multiWCCounter > multiWC) multiWCCounter = 1;
+					while (localCounter < multiWCCounter)
+					{
+						if (*ptr == '_') ++localCounter;
+						++ptr;
+					}
+				}
+				else if (*src == '#' && arrayWC)
+				{
+					++arrayWCCounter;
+					if (arrayWCCounter > arrayWC) arrayWCCounter = 1;
+					while (localCounter < arrayWCCounter)
+					{
+						if (*ptr == '#') ++localCounter;
+						++ptr;
+					}
+				}
+				--ptr;
+				int noMatch = 0;
+				char * ptrCopy = ptr;
+
+				// check if the key name part containing the wildcard matches the globbing key
+
+				// check starting at the wildcard until the start of the name part.
+				while (!noMatch && ptrCopy > relativeName && segmentStart > metaString)
+				{
+					if (*ptrCopy == '?')
+					{
+						while (*segmentStart != '/' && segmentStart > metaString && ptrCopy > relativeName &&
```

Comment:
it's pretty much the same. the first block searches from the wildcard position to the beginning of the keyname part and check if it matches the part from the globbing string, the second block, only if the first succeeded, from the wildcard position to the name part end